### ISSUE TITLE
[Feature Request] Minecart with Crafting Automat 

### ISSUE THREAD

**Issue by BenGoldberg1** [Lolcroc/ProjectLepton#21]
Add a recipe for a "Crafting Automat" and Minecart, producing a "Crafting Automat with Minecart" item.
This item can be placed in the world like any other Minecart or "Minecart with <block>".

Hoppers and droppers can interact with the entity just as they can with the block version of it.

Whenever it is above a powered Activator Rail, it will try to craft one of it's recipe.

The Minecart with Crafting Automat has ten possible directions to spit out the crafting results: north, south, east, west, up, down, left, right, forwards, and backwards.  The output direction is selected either via it's GUI, or by dispensing a compass onto the Minecart to cycle through them.

With the first six, the Crafting Automat within the Minecart maintains a constant direction relative to the world.  With the last four, the Crafting Automat within the Minecart maintains a constant direction relative to the Minecart's direction of travel.

**Comment by mont-40** [comment]
Excellent idea

**Comment by AiDSl** [comment]
I think this should seprate to another mod that also introduce dropper minecart and dispenser,and rather than control shooting dirction using gui,control it using rail with configuration is much better idea.

tho perserve a interface to add crafting minecraft for that could be great indeed.

**Comment by mont-40** [comment]
@AiDSl Yes I agree. It makes sense to have activator rails trigger the automat, with another mechanism possibly switching between locking the direction (north, west, south, east) and unlocking it (forwards/backwards/left/right) based on which direction the minecraft and automat currently are. Perhaps two activator rails behind one another trigger the second mechanism.
